praxis_data/lib/sources/local/praxis_database.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:async';
+
+import 'package:injectable/injectable.dart';
+import 'package:path/path.dart';
+import 'package:sqflite/sqflite.dart';
+
+import '../../models/jokes/dt_joke.dart';
+
+const String _dbName = "praxis.db";
+
+@singleton
+class PraxisDatabase {
+  static Database? _database;
+
+  PraxisDatabase();
+
+  Future<Database> get database async {
+    if (_database != null) return _database!;
+    _database = await _initDB(_dbName);
+    return _database!;
+  }
+
+  Future<Database> _initDB(String filePath) async {
+    final dbPath = await getDatabasesPath();
+    final path = join(dbPath, filePath);
+    return await openDatabase(path, version: 1, onCreate: _createDB);
+  }
+
+  Future _createDB(Database db, int version) async {
+    const idType = "INTEGER PRIMARY KEY AUTOINCREMENT";
+    const jokeType = "TEXT NOT NULL";
+    await db.execute("""
```

Comment:
@shubhamsinghmutualmobile RAW Query 🤔 can we move this as a const in a file ?

Reply:
This method `_createDB` deals with this particular insert DB operation only.. nothing else. Would it be efficient to extract its contents when they're not going to be used anywhere else?